src/include/api.h

```diff
@@ -67,6 +67,12 @@
     (s)->name = __oldname;  \
     --(s)->api_call_counter
 
+#define API_NAME_PUSH(s, struct_name, func_name) \
+    const char *__oldname;                       \
+    __oldname = (s)->name;                       \
```

Comment:
We can keep 
```
/* If this isn't an API reentry, the name should be NULL and the counter should be 0. */
WT_ASSERT(s, (s)->name != NULL || (s)->api_call_counter == 0);
``` 
in API_NAME_PUSH since it'll only be present in diagnostic mode

Reply:
fixed